### DISCUSSION TITLE
Col forno non si scherza...

### DISCUSSION THREAD

**Discussion by Dregi56** [indomus/forum#53] [HUB: Home Assistant]
Ho un forno elettrico in cucina controllato da una presa Tuya e ne sto verificando il funzionamento in riferimento all'opzione _status_ dopo "ritorno corrente". L'impostazione dal cloud Smart Live non mi sembra molto affidabile. Siccome qui si tratta di un forno e non di una lampadina è chiaro che diventa molto importante essere sicuri che al ritorno della corrente dopo una sospensione, lo stato sia off. Ho anche il dubbio che la suddetta opzione non funzioni in quanto al ritorno della corrente deve ripartire il Router il quale deve quindi ricollegarsi alla presa smart, ma analogamente anche Raspberry con HA dovrebbe ripartire....
Può HA ripartendo verificare lo status di quella presa? Penso proprio di si, ma non so da dove cominciare.

**Comment by snakuzzo** [comment]
Premesso che non ho prese Tuya, non so come funziona il cloud di Tuya, posso dirti che:

1. Al tuo posto valuterei il flash con firmware tipo Tasmota, Esphome o simili (a patto che la presa abbia un chip ESP826x. In questo modo controlleresti la presa localmente e non tramite cloud
2. Home Assistant è "OVVIO" che può recuperare lo stato di un device allo start, ma se questa presa è connessa al cloud...quindi deve avviarsi il tuo server (RPI), deve aspettare che il modem permetta di uscire su internet e che la presa si connetta al cloud. Ovviamente il tutto avverrà in quest'ordine...

- HA si avvia
- Il modem sarà connesso
- la presa ed il componente di HA si collegheranno al cloud

Dettò ciò tu stai usando un'integrazione nativa...devi dare un'occhiata a quella e capire se c'è un servizio per forzare o meno il recupero dello stato.

Io personalmente sulle Blitzwolf che ho a casa ho messo Tasmota utilizzando [Tuya Convert](https://github.com/ct-Open-Source/tuya-convert). Le controllo in locale via MQTT e ho lo stato appena si avvia HA (a prescindere dalla connessione ad internet)

**Reply by Giuseppe-P** [reply]
Io personalmente uso loetad_eu3s_plug ed ho messo esphome utilizzando tuya convert. Data l'integrazione nativa con esphome tramite API, HA rileva tranquillamente il suo stato al riavvio

**Reply by snakuzzo** [reply]
Io suggerisco anche Tasmota per il fatto che per chi non è pratico può sembrare più complicato e magari senza il progettino già pronto per poter creare il firmware può avere difficoltà. Ma tra i due firmware non c'è paragone. ESPHome è ad un livello superiore

**Reply by Dregi56** [reply]
Ho anche studiato greco, ma per me quello è arabo! Ho capito che per arrivare dove indicate voi, devo fare ancora molta strada...  :)
